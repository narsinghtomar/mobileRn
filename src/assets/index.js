/**
 * Images
 */

import {default as IMAGES} from './images';

export {IMAGES};
