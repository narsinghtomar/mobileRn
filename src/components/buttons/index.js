/**
 * import button components
 */
import ButtonBlock from './buttonBlock';
import Button from './button';
import IconButton from './iconButton';

/**
 * export
 */
export {ButtonBlock, Button, IconButton};
