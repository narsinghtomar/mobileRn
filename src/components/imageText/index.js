/**
 * ImageWithText
 */
import React from 'react';
import {Text, Image, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {COLOR_SECONDARY} from '../../utils/colors';

const ImageWithText = props => {
  const {
    imagePosition = 'left',
    title,
    titleStyle,
    imgStyle,
    size = 12,
    tintColor = COLOR_SECONDARY,
    resizeMode = 'contain',
    isClickable = false,
    containerStyle = {},
    onPress,
    name,
    image,
  } = props || {};

  const imgSizeColorStyle = {height: size, width: size, tintColor: tintColor};

  return (
    <TouchableOpacity
      disabled={!isClickable}
      style={[styles.parent, containerStyle]}
      onPress={() => {
        onPress && onPress(name);
      }}>
      {imagePosition === 'left' && (
        <Image
          source={image}
          style={[styles.imgLeft, imgSizeColorStyle, imgStyle]}
          resizeMode={resizeMode}
        />
      )}
      <Text style={[styles.title, titleStyle]}>{title}</Text>
      {imagePosition === 'right' && (
        <Image
          source={image}
          style={[styles.imgRight, imgSizeColorStyle, imgStyle]}
          resizeMode={resizeMode}
        />
      )}
    </TouchableOpacity>
  );
};

export default ImageWithText;
