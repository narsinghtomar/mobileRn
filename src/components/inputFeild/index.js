/**
 * import
 */
import InputField from './inputField';

/**
 * export
 */
export {InputField};
