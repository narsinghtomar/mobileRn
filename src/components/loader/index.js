/**
 * Loader
 */
import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import {COLOR_SECONDARY} from '../../utils/colors';
import {styles} from './styles';

const Loader = props => {
  const {isLoading} = props;
  return isLoading ? (
    <View style={styles.loaderContainer}>
      <View style={styles.indicator}>
        <ActivityIndicator
          size={'large'}
          animating={isLoading}
          color={COLOR_SECONDARY}
        />
      </View>
    </View>
  ) : null;
};

export default Loader;
