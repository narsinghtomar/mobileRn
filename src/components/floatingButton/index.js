/**
 * Floating Button
 */
import React from 'react';
import {Image, TouchableOpacity} from 'react-native';
import {IMAGES} from '../../assets';
import {styles} from './styles';

const FloatingButton = props => {
  const {onPress} = props || {};
  return (
    <TouchableOpacity
      onPress={() => onPress && onPress()}
      style={styles.container}>
      <Image source={IMAGES.SEARCH} style={styles.icon} resizeMode="contain" />
    </TouchableOpacity>
  );
};

export default FloatingButton;
