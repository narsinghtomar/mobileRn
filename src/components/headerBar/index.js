/**
 * Header Bar Components.
 */
import React from 'react';
import {TouchableOpacity, Text, View, Image} from 'react-native';
import {getStatusBarHeight} from '../../utils/helper';
import {COLOR_SECONDARY} from '../../utils/colors';
import {IMAGES} from '../../assets';
import {styles} from './styles';
import Space from '../space';

const HeaderBar = props => {
  const {
    navigation,
    headerText = 'Article Details',
    onBackPress,
    isBackIconShow = true,
    backIcon = IMAGES.ARROW,
    callBackFunction,
  } = props || {};

  const backPress = () => {
    callBackFunction && callBackFunction();
    navigation.goBack();
  };

  /**
   * _buildBackIconView
   */
  const _buildBackIconView = () => {
    return (
      <View style={styles.backIconView}>
        {isBackIconShow && (
          <TouchableOpacity
            style={[styles.touchableHighlight]}
            underlayColor={COLOR_SECONDARY}
            onPress={onBackPress ? onBackPress : backPress}>
            <Image source={backIcon} style={styles.icon} resizeMode="contain" />
          </TouchableOpacity>
        )}
      </View>
    );
  };

  /**
   * _buildHeaderTextView
   */
  const _buildHeaderTextView = () => {
    return (
      <View style={styles.textView}>
        <Text style={styles.headerText}>{headerText}</Text>
      </View>
    );
  };

  /**
   *
   */
  return (
    <>
      <Space height={getStatusBarHeight(true)} />
      <View style={styles.headerContainer}>
        {_buildBackIconView()}
        {_buildHeaderTextView()}
      </View>
    </>
  );
};

export default HeaderBar;
