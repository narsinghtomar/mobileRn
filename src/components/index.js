/**
 * Import Components
 */
import {SnackBar, HideSnackBar} from './snackBar';
import Loader from './loader';
import Space from './space';
import {ButtonBlock, Button, IconButton} from './buttons';
import ImageWithText from './imageText';
import {InputField} from './inputFeild';
import HeaderBar from './headerBar';
import FloatingButton from './floatingButton';
/**
 * Export
 */
export {
  SnackBar,
  HideSnackBar,
  Loader,
  Space,
  ButtonBlock,
  Button,
  IconButton,
  ImageWithText,
  InputField,
  HeaderBar,
  FloatingButton,
};
