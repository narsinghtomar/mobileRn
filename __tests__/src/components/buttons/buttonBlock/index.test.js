/**
 * Button
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {ButtonBlock} from '../../../../../src/components';

it('ButtonBlock component renders correctly', () => {
  const tree = renderer.create(<ButtonBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});
