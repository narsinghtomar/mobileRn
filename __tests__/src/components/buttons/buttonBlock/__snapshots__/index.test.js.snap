// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBlock component renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#6F10A2",
      "borderRadius": 4,
      "height": 45,
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 12,
      "width": 675,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "700",
          "letterSpacing": 0.25,
          "textAlign": "center",
          "textAlignVertical": "center",
        },
        undefined,
      ]
    }
  />
</View>
`;
